### PR TITLE
[WIP] - Feature/radio includes

### PR DESCRIPTION
**Pull request by katedee**
This is a base idea around how to handle radio input forms in our express/pug app.

Plenty of room for improvement, I'm sure. I've left some comments in areas about things that may feel clunky/reasoning behind, but am open to suggestions (even if the suggestion is to scrap it)

having issues with one of tests right now, will sort that out later if we feel this is worth going ahead with

The idea is:
- keep the radio form values (which is essentially text for our labels) in a file that can be imported multiple places, since we'll need these values in multiple places (validation, text, etc), and it'd be nice to have one source of truth vs. editing copy in multiple files
- pass these values along to our route, access them in the pug file
- pug file has a var which is an object with the common vals in the radio forms (names for buttons, dataKey which is used to check if the button should be pre-checked on page load), and then a list of values we're using for the radio buttons

### CODE REVIEW

File: views/personal/maritalStatus-edit.pug [hunk 1]
```diff
@@ -10,27 +10,18 @@ block content
   div
     p Please choose the correct status from the options listed below.
 
+  //- this is messy (data && data.personal && data.personal.maritalStatus) but pug will error if a property doesn't exist. We could handle this on the controller level, and pass this to the route? Will help keep it a bit cleaner here
```
Comment (pcraig3) [inline views/personal/maritalStatus-edit.pug]: Yeah, I hate that

File: views/personal/maritalStatus-edit.pug [hunk 2]
```diff
@@ -10,27 +10,18 @@ block content
...
+  - var radioFormOther = {common: { name: 'maritalStatus', dataKey: (data && data.personal && data.personal.maritalStatus ? data.personal.maritalStatus : null) }, options: maritalStatusOptions.en }
```
Comment (pcraig3) [inline views/personal/maritalStatus-edit.pug]: Is `dataKey` the same as the "value" html attribute? (ie, could we call it `value`)

File: formRadios.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// haven't used this elsewhere yet, but the idea of keeping it here is so we can pull these values in wherever we're using them, and have one source of truth for these values (instead of updating copy in multiple files)
+const maritalStatusOptions = {
+    en: [
+      {
+        value: 'Single',
+        label: 'Single',
+      },
+      {
+        value: 'Married',
```
Comment (pcraig3) [inline formRadios.js]: Currently we have this as "Married or common-law"

File: formRadios.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      }, 
+      {
+        value: 'Widowed',
+        label: 'Widowed',
+      }
+    ]
+}
+module.exports = {
+  maritalStatusOptions
+}
```
Comment (pcraig3) [inline formRadios.js]: HMMMMM.

Can we put these in the schemas file somewhere? 
That way everything to do with questions will live together.

Also, the `en`, `fr` thing I am a bit suspicious of (all of our other content will work with the translate lib right now), but happy to let this through if you think it's easier than the alternative (ie, like what we've done with error messages). 

Finally, we wouldn't want to change the value in French, only the key.

Comment (pcraig3) [reply formRadios.js]: If you wanted to keep it in its own file, you could do something like I did with the /utils file recently. (ie, move utils.js to /utils/index.js)

https://github.com/cds-snc/cra-claim-tax-benefits/pull/25/files#diff-8d6b1f9270334e293ad1f2ddec84e050

File: views/personal/maritalStatus-edit.pug (hunk 2 above)
Comment (katedee) [reply views/personal/maritalStatus-edit.pug]: so datakey is checking the data as we enter the page it's basically used to run the check:
`checked=(radioForm.common.dataKey &&radioForm.common.dataKey.toLowerCase()==option.value.toLowerCase()))` to see if the item should be checked when we load the page (ex. someone updating their marital status where previously it was 'Single' and so when they get to the page 'Single' will be checked before they change it).

We could get rid of pre-checking altogether, or handle it differently! I'm open to alternatives as making sure that first `data` exists, and then `data.personal` exists, and then `data.personal.maritalStatus` exists. We could offload this series of checks into a function that's in utils?

File: formRadios.js (hunk 3 above)
Comment (katedee) [reply formRadios.js]: will update!

File: views/personal/maritalStatus-edit.pug (hunk 1 above)
Comment (sboisvert) [reply views/personal/maritalStatus-edit.pug]: >Will help keep it a bit cleaner here

Trying to keep the view as clean as possible seems reasonable.

File: views/personal/maritalStatus-edit.pug [hunk 5]
```diff
@@ -10,27 +10,18 @@ block content
...
-                    label.radios__label(for="maritalStatusSeparated") Separated
-                .radios__item
-                    input.radios__input(id="maritalStatusDivorced" name="maritalStatus" type="radio" value="Divorced" checked=(data && data.personal.maritalStatus.toLowerCase()=='divorced') aria-describedby=(errors ? 'maritalStatus_error' : false))
-                    label.radios__label(for="maritalStatusDivorced") Divorced
-                .radios__item
-                    input.radios__input(id="maritalStatusWidowed" name="maritalStatus" type="radio" value="Widowed" checked=(data && data.personal.maritalStatus.toLowerCase()=='widowed') aria-describedby=(errors ? 'maritalStatus_error' : false))
-                    label.radios__label(for="maritalStatusWidowed") Widowed
+        
+            include ../_includes/radioButtons.pug
+            +radio-buttons(radioFormOther)
```
Comment (sboisvert) [inline views/personal/maritalStatus-edit.pug]: What does `radioFormOther` reference?

**Pull request closed without merging**